src/common/stats/MetricsSerializer.h

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include "StatsManager.h"
+
+namespace nebula {
+namespace stats {
+
+class MetricsSerializer {
+public:
+    virtual ~MetricsSerializer() = default;
+    std::string serialize(StatsManager& sm) const {
+        std::ostringstream ss;
+        serialize(ss, sm);
+        return ss.str();
+    }
+    virtual void serialize(std::ostream& out, StatsManager& sm) const = 0;
+};
+
+class PrometheusSerializer : public MetricsSerializer {
+public:
+    virtual ~PrometheusSerializer() = default;
+    void serialize(std::ostream& out, StatsManager& sm) const override {
+        prometheusSerialize(out, sm);
+    }
+    void prometheusSerialize(std::ostream& out, StatsManager& sm) const;
```

Comment:
Why not override the serialize virtual function of MetricsSerializer directly? Why to define the new function prometheusSerialize and proxy to it?

Reply:
No special reason, just for more clear.